qmail.c

```diff
@@ -88,10 +98,21 @@ struct qmail *qq;
 {
   int wstat;
   int exitcode;
+  char ch;
+  static char errstr[256];
+  size_t len = 0;
 
   qmail_put(qq,"",1);
   if (!qq->flagerr) if (substdio_flush(&qq->ss) == -1) qq->flagerr = 1;
   close(qq->fde);
+  substdio_fdbuf(&qq->ss,read,qq->fderr,qq->buf,sizeof(qq->buf));
+  while (substdio_bget(&qq->ss,&ch,1) && len < 255) {
```

Comment:
We should clean this up a little bit, the ch variable isn't actually needed. One could pass ```errstr + len``` as second argument and just count len up.

Reply:
Like so?

```c
while (substdio_bget(&qq->ss,errstr+len++,1) && len < 256);
errstr[len] = '\0';
```

Reply:
Also, what's actually the difference between `substdio_bget()` and `substdio_get()`? I see lots of the latter reading a byte at a time and am wondering why we're different.

Reply:
The length they pass to oneread() is the only real difference.

I would write the code as:

```c
while (substdio_bget(&qq->ss,errstr+len,1) && len < 256) {
  len++;
}
errstr[len] = '\0';
```

Otherwise len is one larger than before, and I see this as more readable.

Reply:
Agreed. I think even better if I extract a little function.

There must be some a trick to express `substdio_get` in terms of `substdio_bget`, or vice versa, or at least have most of the work done by a static function they both call, but that doesn't need to be in this PR (or in 1.09).